src/analysis/max_min_driving_force.jl

```diff
@@ -131,8 +171,164 @@ function max_min_driving_force(
     is_solved(opt_model) || return nothing
 
     return (
-        mmdf = objective_value(opt_model),
-        dgs = Dict(rid => value(dgs[i]) for (i, rid) in enumerate(reactions(model))),
-        cs = Dict(mid => exp(value(logcs[i])) for (i, mid) in enumerate(mids)),
+        mmdf = value(opt_model[:mmdf]),
+        dg_reactions = Dict(
+            rid => value(opt_model[:dgrs][i]) for (i, rid) in enumerate(reactions(model))
+        ),
+        concentrations = Dict(
+            mid => exp(value(opt_model[:logcs][i])) for
+            (i, mid) in enumerate(metabolites(model))
+        ),
+    )
+end
+
+"""
+    max_min_driving_force_variability(
+        model::MetabolicModel,
+        reaction_standard_gibbs_free_energies::Dict{String,Float64},
+        optimizer;
+        workers =[myid()],
+        relax_driving_force = 0.9,
+        flux_solution::Dict{String,Float64} = Dict{String,Float64}(),
+        proton_ids::Vector{String} = ["h_c", "h_e"],
+        water_ids::Vector{String} = ["h2o_c", "h2o_e"],
+        constant_concentrations::Dict{String,Float64} = Dict{String,Float64}(),
+        concentration_ratios::Dict{Tuple{String,String},Float64} = Dict{
+            Tuple{String,String},
+            Float64,
+        }(),
+        concentration_lb = 1e-9,
+        concentration_ub = 100e-3,
+        T = _constants.T,
+        R = _constants.R,
+        small_flux_tol = 1e-6,
+        modifications = [],
+        ignore_reaction_ids = [],
     )
+
+Perform a variant of flux variability analysis on a max min driving force type problem. All
+the arguments are forwarded to [`max_min_driving_force`](@ref). Calls [`screen`](@ref)
+internally and possibly distributes computation across `workers`. First performs regular max
+min driving force analysis to find the max min driving force of the model. Then iteratively
+maximizes and minimizes the driving force across each reaction, and then the concentrations
+while staying close to the original max min driving force. Specifically, the max min driving
+force is constrained to be beween the optimum, as found using
+[`max_min_driving_force`](@ref), and `relax_driving_force` times the optimum. If
+`relax_driving_force = 1`, then the max min driving force of the problem is fixed to that
+value instead of constrained.
+
+Returns a matrix of solutions to [`max_min_driving_force`](@ref) additionally constrained as
+described above, where the rows are in the order of the reactions and then the metabolites
+of the `model`. For the reaction rows the first column is the maximum dG of that reaction,
+and the second column is the minimum dG of that reaction subject to the above constraints.
+For the metabolite rows, the first column is the maximum concentration, and the second column
+is the minimum concentration subject to the constraints above.
+"""
+function max_min_driving_force_variability(
+    model::MetabolicModel,
+    reaction_standard_gibbs_free_energies::Dict{String,Float64},
```

Comment:
parameters:
- for similarity with FVA it would be better to support a reaction range
- `reaction_standard_gibbs_free_energies` is a bit long but might be better as a named arg.
- use `kwargs...` for the rest

Reply:
fixed everything, but kept the longer name - in the docstring it is quite informative